File: CS/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using DevExpress.XtraGrid.Columns;
using System.Collections;

namespace CreateGridSplitContainer {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {

            IList dataSource = CreateDataSource();

            GridSplitContainer gridSplitContainer = new GridSplitContainer();
            gridSplitContainer.Parent = this;
            gridSplitContainer.Location = new Point(0, 0);
            gridSplitContainer.Size = new Size(350, 300);
            gridSplitContainer.Initialize();
            gridSplitContainer.SplitViewCreated += new EventHandler(gridSplitContainer_SplitViewCreated);

            // Customize the first grid control.
            GridControl grid = gridSplitContainer.Grid;
            GridView view = grid.MainView as GridView;
            // Specify a data source.
            grid.DataSource = dataSource;
            // Resize columns according to their values.
            view.BestFitColumns();
            // Locate a row containing a specific value.
            view.FocusedRowHandle = view.LocateByValue("Country", "UK");

            // Display a splitter and second grid control.
            gridSplitContainer.ShowSplitView();

            // Customize the second grid control.
            GridControl secondGrid = gridSplitContainer.SplitChildGrid;
            GridView secondView = secondGrid.MainView as GridView;
            // Locate a row containing a specific value.
            secondView.FocusedRowHandle = secondView.LocateByValue("Country", "Sweden");
        }

        private void gridSplitContainer_SplitViewCreated(object sender, EventArgs e) {
            // Display the Embedded Navigator for the second grid control 
            // in the horizontally oriented Split View.
            GridSplitContainer gsc = sender as GridSplitContainer;
            if (!gsc.Horizontal)
                gsc.SplitChildGrid.UseEmbeddedNavigator = true;
        }


        private IList CreateDataSource() {
            List<MyRecord> list = new List<MyRecord>();
            list.Add(new MyRecord(1, "Rene Phillips", "USA"));
            list.Add(new MyRecord(2, "Robert McKinsey", "UK"));
            list.Add(new MyRecord(3, "Christina Berglund", "Sweden"));
            list.Add(new MyRecord(4, "Martín Sommer", "Spain"));
            list.Add(new MyRecord(5, "Laurence Lebihan", "France"));
            list.Add(new MyRecord(6, "Elizabeth Lincoln", "Canada"));
            list.Add(new MyRecord(7, "Steven Baum", "USA"));
            return list;
        }

        
    }

    public class MyRecord {
        public int ID { get; set; }
        public string Country { get; set; }
        public string Name { get; set; }
        public MyRecord(int id, string name, string country) {
            ID = id;
            Name = name;
            Country = country;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports DevExpress.XtraGrid.Columns
Imports System.Collections

Namespace CreateGridSplitContainer

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            Dim dataSource As IList = CreateDataSource()
            Dim gridSplitContainer As GridSplitContainer = New GridSplitContainer()
            gridSplitContainer.Parent = Me
            gridSplitContainer.Location = New Point(0, 0)
            gridSplitContainer.Size = New Size(350, 300)
            gridSplitContainer.Initialize()
            gridSplitContainer.SplitViewCreated += New EventHandler(AddressOf gridSplitContainer_SplitViewCreated)
            ' Customize the first grid control.
            Dim grid As GridControl = gridSplitContainer.Grid
            Dim view As GridView = TryCast(grid.MainView, GridView)
            ' Specify a data source.
            grid.DataSource = dataSource
            ' Resize columns according to their values.
            view.BestFitColumns()
            ' Locate a row containing a specific value.
            view.FocusedRowHandle = view.LocateByValue("Country", "UK")
            ' Display a splitter and second grid control.
            gridSplitContainer.ShowSplitView()
            ' Customize the second grid control.
            Dim secondGrid As GridControl = gridSplitContainer.SplitChildGrid
            Dim secondView As GridView = TryCast(secondGrid.MainView, GridView)
            ' Locate a row containing a specific value.
            secondView.FocusedRowHandle = secondView.LocateByValue("Country", "Sweden")
        End Sub

        Private Sub gridSplitContainer_SplitViewCreated(ByVal sender As Object, ByVal e As EventArgs)
            ' Display the Embedded Navigator for the second grid control 
            ' in the horizontally oriented Split View.
            Dim gsc As GridSplitContainer = TryCast(sender, GridSplitContainer)
            If Not gsc.Horizontal Then gsc.SplitChildGrid.UseEmbeddedNavigator = True
        End Sub

        Private Function CreateDataSource() As IList
            Dim list As List(Of MyRecord) = New List(Of MyRecord)()
            list.Add(New MyRecord(1, "Rene Phillips", "USA"))
            list.Add(New MyRecord(2, "Robert McKinsey", "UK"))
            list.Add(New MyRecord(3, "Christina Berglund", "Sweden"))
            list.Add(New MyRecord(4, "Martín Sommer", "Spain"))
            list.Add(New MyRecord(5, "Laurence Lebihan", "France"))
            list.Add(New MyRecord(6, "Elizabeth Lincoln", "Canada"))
            list.Add(New MyRecord(7, "Steven Baum", "USA"))
            Return list
        End Function
    End Class

    Public Class MyRecord

        Public Property ID As Integer

        Public Property Country As String

        Public Property Name As String

        Public Sub New(ByVal id As Integer, ByVal name As String, ByVal country As String)
            Me.ID = id
            Me.Name = name
            Me.Country = country
        End Sub
    End Class
End Namespace
